CHANGELOG.md

```diff
@@ -4,6 +4,11 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
```

Comment:
good to have now additional info on top, overal giving us more flexibility to polish these pages and make them actually useful

you've pointed out the downsides of this update in the PR description

is there any merit to simply provide the text now at the top of the file explicitly while simply generating the toctree below automatically?

Seems to me that hybrid would be best solution. I don't see the argument for more flexibility in the names etc, we might as well change the .py file names in such cases

Reply:
So you propose to have something like `Basics_Header.md` which just contains the text that is added at the top of the file and create everything else as before, just including the text from `Basic_Header.md`?
Would be fine for me, I do not have a strong opinion here.

Reply:
yeah it would be best of both worlds, no?

Reply:
@AVHopp: the hybrid solution is also actually what I had in mind. So it turns out we were talking about slightly different things in the meeting 😄 Can you adjust it?

Reply:
Your wishes are my orders. 😉

Reply:
I will however do a rebase then and also include the other comments currently present here and notify you once it is "safe" to review again.

Reply:
@AdrianSosic @Scienfitz : Change included. Please have a look if this is to your liking and resolve if gucci.

Reply:
The formatting is weird for some reason. Any idea?
<img width="759" alt="image" src="https://github.com/emdgroup/baybe/assets/23265127/7e65fb4f-eb95-48a7-92ed-882b822ae408">

Reply:
whats strange here?

Reply:
The increased whitespace of the line break. Also appears in the other cases. In the examples themselves, everything is back to normal

Reply:
I guess that this has to do with the way that I "glue" the lines of the headers together. I will check, should be something rather minor.

Reply:
I have changed it by removing a manually entered new line. This now results in all **paragraphs** in the header files being rendered as one chunk of text without new lines, even if there are new lines in the markdown file (basically "LaTeX style"). Please have a look locally (and also verify that this is as you want to have it!) and confirm.

Reply:
Well, why no just copying the text AS IS into the markdown file? Because when you wrote the markdown file yourself, you would also want a blank line to appear as a blank line?

Reply:
Ahhh, now I see where the confusion lies. Yeah, your `\n`-joining was the issue. And btw, to reduce your pain: `readlines` reads the file but, as you said, line by line. If your only purpose of calling it is to then just rejoin the lines, you can just skip the line-by-line approach and read the file at once, using `read`. But since you are already using a `Path` object, you can even omit the context creation step and just call `.read_text()`. Fixed it and pushed the commit 👍🏼